ntoskrnl/mm/ARM3/mminit.c

```diff
@@ -2324,10 +2325,9 @@ MmArmInitSystem(IN ULONG Phase,
         }
 
         /* Subsection PTEs are always in nonpaged pool, pick a random address to try */
-        PointerPte = (PMMPTE)((ULONG_PTR)MmNonPagedPoolStart + (MmSizeOfNonPagedPoolInBytes / 2));
-        MI_MAKE_SUBSECTION_PTE(&TempPte, PointerPte);
-        TestPte = MiSubsectionPteToSubsection(&TempPte);
-        ASSERT(PointerPte == TestPte);
+        Subsection = (PSUBSECTION)((ULONG_PTR)MI_NONPAGED_POOL_END - _1MB);
```

Comment:
Why this change?

Reply:
It could be like this:
PSUBSECTION MiSubsectionPteToSubsection(_In_ PMMPTE SubsectionPte);
But since besides SUBSECTION there is also MSUBSECTION, then:
PVOID MiSubsectionPteToSubsection(_In_ PMMPTE SubsectionPte);
And SUBSECTION (or MSUBSECTION) is not MMPTE at all ...